test/bun.js/util/util-promisify.test.js

```diff
@@ -0,0 +1,81 @@
+import { describe, it } from "bun:test";
+import fs from 'node:fs';
+import { promisify } from 'util';
+import assert from "assert";
+
+const stat = promisify(fs.stat);
+
+describe("util.promisify", () => {
+  describe("promisify fs calls", () => {
+    // TODO: common.mustCall is not implemented here yet
+    // https://github.com/nodejs/node/blob/main/test/common/index.js#L398
+    it.skip("all cases", () => {
+      const promise = stat(__filename);
+      assert.equal(promise instanceof Promise, true);
+      promise.then(common.mustCall((value) => {
+        assert.deepStrictEqual(value, fs.statSync(__filename));
+      }));
+
+
+      const promiseFileDontExist = stat('/dontexist');
+      promiseFileDontExist.catch(common.mustCall((error) => {
+        assert(error.message.includes('ENOENT: no such file or directory, stat'));
+      }));
+    })
+  })
+  
+  describe("promisify.custom", () => {
+    it("double promisify", () => {
+      function fn() {}
+
+      function promisifedFn() {}
+      fn[promisify.custom] = promisifedFn;
+      assert.strictEqual(promisify(fn), promisifedFn);
+      assert.strictEqual(promisify(promisify(fn)), promisifedFn);
+    })
+
+    // TODO: remove skip when shared symbol is test-able
+    it.skip("should register shared promisify symbol", () => {
+      function fn() {}
+
+      function promisifiedFn() {}
+
+      // TODO: register shared symbol promisify.custom
+      // util.promisify.custom is a shared symbol which can be accessed
+      // as `Symbol.for("nodejs.util.promisify.custom")`.
+      const kCustomPromisifiedSymbol = Symbol.for('nodejs.util.promisify.custom');
+      fn[kCustomPromisifiedSymbol] = promisifiedFn;
+
+      assert.strictEqual(kCustomPromisifiedSymbol, promisify.custom);
+      assert.strictEqual(promisify(fn), promisifiedFn);
+      assert.strictEqual(promisify(promisify(fn)), promisifiedFn);
+    })
+  })
+
+  it("should fail when type is not a function", () => {
+    function fn() {}
+    fn[promisify.custom] = 42;
+    assert.throws(
+      () => promisify(fn),
+      // TODO: error code is not the same as node's.
+      // { code: 'ERR_INVALID_ARG_TYPE', name: 'TypeError' }
+      { name: 'TypeError' }
+    );
+  })
+  
+  // TODO: common.mustCall
```

Comment:
Up to this point I decided to stop working on this test for now since it requires common.mustCall to be implemented for further testing.

Should I do this first?

Reply:
I think it will be valuable to land more tests into one PR and address the gaps as a follow up issue.
We will get a better understanding of what are the gaps by that point of time.

Reply:
Another idea is to use similar strategy as used here
https://github.com/oven-sh/bun/blob/463f2b7dabcc3fa9d90a14b88ce81262abdd71c7/test/bun.js/setTimeout.test.js#L38
as an alternative of `common.mustCall`

Reply:
hmm I see, I'll see what I can do with that. Thank you for the reference!